src/credentials/GroupDataProviderImpl.cpp

```diff
@@ -702,6 +702,17 @@ CHIP_ERROR GroupDataProviderImpl::GetGroupInfo(chip::FabricIndex fabric_index, c
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR GroupDataProviderImpl::RemoveGroupInfo(chip::FabricIndex fabric_index, chip::GroupId group_id)
+{
+    FabricData fabric(fabric_index);
+    GroupData group;
+
+    VerifyOrReturnError(CHIP_NO_ERROR == fabric.Load(mStorage), CHIP_ERROR_INVALID_FABRIC_ID);
```

Comment:
This seems to convert any  error from loading to INVALID_FABRIC_ID ... that does not seem correct.

Why not `ReturnErrorOnFailure`  ? or have a way to really validate the fabric id.

Reply:
This condition assumes that the only reason to fail is that the fabric is not found. I agree it is a poor assumption, so I have improved these checks so the actual error is returned to the caller function.
Now CHIP_ERROR_INVALID_FABRIC_ID is returned only if the fabric_index is actually invalid, and CHIP_ERROR_NOT_FOUND if the given entry is not found.